pkg/ebpf/c/tracee.bpf.c

```diff
@@ -967,10 +986,40 @@ BPF_LRU_HASH(bpf_attach_tmp_map, u32, bpf_attach_t, 1024);         // temporaril
 
 // EBPF PERF BUFFERS -------------------------------------------------------------------------------
 
+BPF_PERF_OUTPUT(errors, 1024);      // errors submission
 BPF_PERF_OUTPUT(events, 1024);      // events submission
 BPF_PERF_OUTPUT(file_writes, 1024); // file writes events submission
 BPF_PERF_OUTPUT(net_events, 1024);  // network events submission
 
+// HELPERS: ERRORS ---------------------------------------------------------------------------------
+
+static __always_inline void do_submit_error(void *ctx, struct bpf_error *err, void *func, u32 line)
+{
+    if (!ctx || !err || !func || !line)
+        return;
+
+    bpf_probe_read_kernel_str((void *) &err->_func, sizeof(err->_func), func);
+    err->_func[MAX_ERROR_FUNC_LEN - 1] = '\0';
+    err->_line = line;
+
+    bpf_perf_event_output(ctx, &errors, BPF_F_CURRENT_CPU, err, sizeof(*err));
+}
+
+#define submit_error(ctx, err)                                                                     \
+    do {                                                                                           \
+        do_submit_error(ctx, err, (void *) __FUNCTION__, __LINE__);                                \
+    } while (0);
```

Comment:
Why is this in a `do while` and not just a simple statement?

Reply:
Ah sure. Remainder of the submit_new_error that has multiple lines. Getting rid of the do while wrapper for this. Tks.